src/sora_video_encoder_factory.cpp

```diff
@@ -125,12 +125,15 @@ SoraVideoEncoderFactory::CreateVideoEncoder(
     const webrtc::SdpVideoFormat& format) {
   if (internal_encoder_factory_ != nullptr) {
     // サイマルキャストの場合はアダプタを噛ましつつ、無条件ですべてアライメントする
-    auto encoder = std::make_shared<webrtc::SimulcastEncoderAdapter>(
-        internal_encoder_factory_.get(), format);
-    // kNative を I420 に変換する
-    auto adapted1 = std::make_shared<I420EncoderAdapter>(encoder);
-    auto adapted2 = absl::make_unique<AlignedEncoderAdapter>(adapted1, 16, 16);
-    return adapted2;
+    std::shared_ptr<webrtc::VideoEncoder> encoder =
+        std::make_shared<webrtc::SimulcastEncoderAdapter>(
+            internal_encoder_factory_.get(), format);
+
+    if (config_.force_simulcast_i420_conversion) {
+      encoder = std::make_shared<I420EncoderAdapter>(encoder);
+    }
+
+    return absl::make_unique<AlignedEncoderAdapter>(encoder, 16, 16);
```

Comment:
同じアダプタなのに最後の１個だけ別の書き方をするのは変なので、同じような書き方をしてあげた方がいい感じに見えます
```suggestion
    encoder = absl::make_unique<AlignedEncoderAdapter>(encoder, 16, 16);
    return encoder;
```

Reply:
ありがとうございます
たしかに、順番を変えたり、アダプタを一番最後に追加したい場合など、提案いただいた書き方が良さそうですね